### PR TITLE
[CSL 8] Add Conversion Types to Tracking

### PR DESCRIPTION
**Pull request by jjl014**
#### Updates:
* Update trackConversion to use v2 endpoint
* Add support for converstion types
* Add/Update tests

### CODE REVIEW

File: src/modules/tracker.js [hunk 1]
```diff
@@ -356,37 +356,66 @@ class Tracker {
...
+      if (is_custom_type) {
+        bodyParams.is_custom_type = is_custom_type;
+      }
+
+      if (display_name) {
+        bodyParams.display_name = display_name;
       }
 
-      this.requests.queue(`${url}${applyParamsAsString(queryParams, this.options)}`);
+      const requestURL = `${requestPath}${applyParamsAsString({}, this.options)}`;
```
Comment (crgee1) [inline src/modules/tracker.js:410]: are we not adding the item_name and item_id still to the request url?

**Review by crgee1 — CHANGES_REQUESTED** [review]
looks cool, left one comment, and if you could update the docs for the new params

File: src/modules/tracker.js (hunk 1 above)
Comment (jjl014) [reply src/modules/tracker.js]: Updated to include it 👍

File: spec/src/modules/search.js [hunk 2]
```diff
@@ -292,24 +291,6 @@ describe('ConstructorIO - Search', () => {
       });
     });
 
-    it('Should return a response from collection with a valid query and collection id', (done) => {
```
Comment (deomsj) [inline spec/src/modules/search.js:295]: why are we removing these collection tests? Are these no longer supported?

**Comment by deomsj** [comment]
Is it an accurate summary to say that when tracking conversion events that have `is_custom_type: true`:
 - `display_name` is required
 - `type` is optional

**Review by crgee1 — APPROVED** [review]
lgtm

File: spec/src/modules/search.js (hunk 2 above)
Comment (jjl014) [reply spec/src/modules/search.js:295]: Yea, the search endpoint no longer supports collection ids. It just returns an error:

`"Search within a collection is not supported."`

**Comment by jjl014** [comment]
> Is it an accurate summary to say that when tracking conversion events that have is_custom_type: true:
> * display_name is required
> * type is optional 

@deomsj Yup, that is correct. Although I'm not entirely sure what happens in that case. (i.e. what is used for the for the type? does it default to `add_to_cart`?). Will followup w/ Wilson to get more details.

**Comment by sblaurock** [comment]
@jjl014 will this break any existing implementations that we have in beacons for customers?

**Pull request merged**